src/main/java/com/api/controller/RocketController.java

```diff
@@ -1,66 +1,44 @@
 package com.api.controller;
 
-import com.api.dto.Rocket_DTO;
-import com.api.dto.RocketMongoId_DTO;
+import com.api.dto.RocketDto;
+import com.api.entity.SxRocket;
+import com.api.mapper.RocketMapper;
+import com.api.service.NetworkingService;
+import com.api.service.RocketPresenter;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
-import org.springframework.web.client.RestTemplate;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping
 public class RocketController {
 
-    private static final String BASE_URL = "https://api.spacexdata.com/v3/rockets";
-    private final RestTemplate restTemplate;
+    private final RocketPresenter rocketPresenter;
 
-    public RocketController(RestTemplate restTemplate) {
-        this.restTemplate = restTemplate;
+    public RocketController(RocketPresenter rocketPresenter) {
+        this.rocketPresenter = rocketPresenter;
     }
 
     @GetMapping
-    public List<Rocket_DTO> getAllRockets(
+    public Mono<List<RocketDto>> getAllRockets(
```

Comment:
Controller should not return a Mono, please return Response with a list of RocketDto